JavaScript/JavaScriptSDK/Sender.ts

```diff
@@ -215,6 +281,57 @@ module Microsoft.ApplicationInsights {
             }
         }
 
+        /** Calculates the time to wait before retrying in case of an error based on
+         * http://en.wikipedia.org/wiki/Exponential_backoff
+         * @param headerValue {string} - Optional header "Retry-After" response header value.
+         */
+        private _setRetryTime(headerValue?: string) {
+            var retryAfterTimeSpan = Date.parse(headerValue);
+            if (!(headerValue && retryAfterTimeSpan > 0)) {
+                const SlotDelayInSeconds = 10;
+                var delayInSeconds: number;
+
+                if (this._consecutiveErrors <= 1) {
+                    delayInSeconds = SlotDelayInSeconds;
+                } else {
+                    var backOffSlot = (Math.pow(2, this._consecutiveErrors) - 1) / 2;
+                    var backOffDelay = Math.floor(Math.random() * backOffSlot * SlotDelayInSeconds) + 1;
+                    delayInSeconds = Math.max(Math.min(backOffDelay, 3600), SlotDelayInSeconds);
+                }
+
+                // TODO: Log the backoff time like the C# version does.
+                retryAfterTimeSpan = Date.now() + (delayInSeconds * 1000);
+            }
+
+            // TODO: Log the retry at time like the C# version does.
+            this._retryAt = retryAfterTimeSpan; 
+        }
+
+        /**
+         * Parses the response from the backend. 
+         * @param response - XMLHttpRequest or XDomainRequest response
+         */
+        private _parseResponse(response: any, retryAfter?: string): IBackendResponse {
+            try {
+                var result = JSON.parse(response);
+
+                if (result && result.itemsReceived && result.itemsReceived >= result.itemsAccepted &&
+                    result.itemsReceived - result.itemsAccepted == result.errors.length) {
+                    return {
```

Comment:
What I mean is that you call JSON.parse(response) which either fails or gives you an object with those fields and then you validate that those expected fields exist.  So why create a new object in the return statement which is basically a copy of the object you already have in the result variable from the original JSON.parse call?